x-pack/plugins/enterprise_search/server/routes/app_search/credentials.test.ts

```diff
@@ -61,6 +170,123 @@ describe('credentials routes', () => {
     });
   });
 
+  describe('PUT /api/app_search/credentials/{name}', () => {
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({ method: 'put', payload: 'body' });
+
+      registerCredentialsRoutes({
+        ...mockDependencies,
+        router: mockRouter.router,
+      });
+    });
+
+    it('creates a request to enterprise search', () => {
+      const mockRequest = {
+        params: {
+          name: 'abc123',
+        },
+      };
+
+      mockRouter.callRoute(mockRequest);
+
+      expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
+        path: '/as/credentials/abc123',
+      });
+    });
+
+    describe('validates', () => {
+      describe('admin keys', () => {
```

Comment:
Note: this is copied/pasted from the new POST route tests. In theory we could DRY it out (since both routes inherit the same exact schema), but I didn't want to over-complicate it for now.